sport/app/football/containers/FixturesAndResults.scala

```diff
@@ -34,13 +36,12 @@ object CompetitionAndGroup {
 
 case class CompetitionAndGroup(competition: Competition, group: Group)
 
-object FixturesAndResults extends Football {
+class FixturesAndResults(competitions: Competitions) extends Football {
   def makeContainer(tagId: String)(implicit request: RequestHeader) = {
-    val competitions = Competitions()
 
     (for {
       teamId <- TeamMap.findTeamIdByUrlName(tagId)
-      teamName <- TeamName.fromId(teamId)
+      teamName <- new TeamNameBuilder(competitions).withId(teamId)
```

Comment:
\<Nit-pick\> `new TeamNameBuilder(competitions)` could be extracted into a private property \</Nit-pick>